boa_engine/src/builtins/intl/mod.rs

```diff
@@ -653,3 +653,141 @@ fn resolve_locale(
+        .unwrap_or_else(|_| JsValue::undefined());
+
+    // 3. If value is undefined, return fallback.
+    if value.is_undefined() {
+        return Ok(fallback.clone());
+    }
+
+    // 4. Assert: type is "boolean" or "string".
+    if r#type.ne("boolean") && r#type.ne("string") {
+        return context.throw_type_error("GetOption: type should be either 'boolean' or 'string'");
+    }
+
+    // 5. If type is "boolean", then
+    if r#type.eq("boolean") {
+        // a. Set value to ! ToBoolean(value).
+        value = JsValue::Boolean(value.to_boolean());
+    }
+
+    // 6. If type is "string", then
+    if r#type.eq("string") {
+        // a. Set value to ? ToString(value).
+        value = JsValue::String(
+            value
+                .to_string(context)
+                .expect("GetOption: failed to convert value to string"),
+        );
+    }
+
+    // 7. If values is not undefined and values does not contain an element equal to value,
+    // throw a RangeError exception.
+    if !values.is_empty() && !values.contains(&value) {
+        return context.throw_range_error("GetOption: values array does not contain value");
+    }
+
+    // 8. Return value.
+    Ok(value)
+}
+
+/// The abstract operation `GetNumberOption` extracts the value of the property named `property`
+/// from the provided `options` object, converts it to a `Number value`, checks whether it is in
+/// the allowed range, and fills in a `fallback` value if necessary.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma402/#sec-getnumberoption
+#[cfg(test)]
+pub(crate) fn get_number_option(
+    options: &JsValue,
+    property: &str,
+    minimum: &JsValue,
+    maximum: &JsValue,
+    fallback: &JsValue,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    // FIXME untested
+    // 1. Assert: Type(options) is Object.
+    if !options.is_object() {
+        return context.throw_type_error("GetOption: options should be an Object");
+    }
+
+    let options_obj = options
+        .to_object(context)
+        .expect("GetOption: options should be an Object");
+
+    // 2. Let value be ? Get(options, property).
+    let value = options_obj
+        .get(property, context)
+        .unwrap_or_else(|_| JsValue::undefined());
+
+    // 3. Return ? DefaultNumberOption(value, minimum, maximum, fallback).
+    default_number_option(&value, minimum, maximum, fallback, context)
+}
+
+/// The abstract operation `DefaultNumberOption` converts `value` to a `Number value`, checks
+/// whether it is in the allowed range, and fills in a `fallback` value if necessary.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma402/#sec-defaultnumberoption
+#[cfg(test)]
+pub(crate) fn default_number_option(
+    value: &JsValue,
+    minimum: &JsValue,
+    maximum: &JsValue,
+    fallback: &JsValue,
+    context: &mut Context,
+) -> JsResult<JsValue> {
```

Comment:
Applied, but I'm not quite sure what to do when the value is undefined and the fallback is None https://github.com/boa-dev/boa/pull/2064/files#diff-1dd4af22d3a6142a8060d056e9d0489d84d8a4d9c68a8f6e7b359b8d3ab35499R759

Reply:
True. I think the best thing would probably be returning a `JsResult<Option<f64>>` and calling `expect` on the `Option` at call site if we are sure it can't be `None` based on the input.